Extract validation from controller

nickdavies791 opened the pull request:
Extracts validation into form request classes `StoreTask `and `UpdateTask `and passes the `$request` data to the job dispatcher. Tidies up comments.

nickdavies791 committed:
Update comments

Updates comments to provide @param and @return values for jobs and
TaskController.

nickdavies791 committed:
Extract validation to Form Request classes

Extracts the validation from the TaskController into their own
respective form request classes. Allows controller to only be
responsible for either dispatching the job to update or save a new task.
Instead of passing array values from the validation it will pass the
request data.

LucasLeandro1204 commented on phpunit.xml:
```diff
@@ -29,6 +29,6 @@
         <env name="MAIL_DRIVER" value="array"/>
         <env name="DB_CONNECTION" value="sqlite"/>
-        <env name="DB_DATABASE" value=":memory:"/>
+        <env name="DB_DATABASE" value="database.testing.sqlite"/>
```
Please rollback this
sqlite running in memory is faster

LucasLeandro1204 commented on app/Jobs/UpdateTask.php:
```diff
@@ -26,7 +26,8 @@ class UpdateTask
      *
-     * @return void
+     * @param Task $task
+     * @param array $attributes
```
you can drop this annotations since we are type hinting the properties

LucasLeandro1204 commented on app/Jobs/DeleteTask.php:
```diff
@@ -19,7 +19,7 @@ class DeleteTask
      * Create a new job instance.
      *
-     * @return void
+     * @param Task $task
```
same

LucasLeandro1204 commented on app/Jobs/CreateTask.php:
```diff
@@ -33,7 +33,9 @@ class CreateTask
-     * @return void
+     * @param string $title
+     * @param string|null $description
+     * @param bool $status
```
same

LucasLeandro1204 commented on app/Http/Requests/UpdateTask.php:
```diff
@@ -0,0 +1,32 @@
+     *
+     * @return array
+     */
+    public function rules()
```
Use type hint instead of annotation

```suggestion
    public function rules(): array
```

LucasLeandro1204 commented on app/Http/Requests/UpdateTask.php:
```diff
@@ -0,0 +1,32 @@
+     *
+     * @return bool
+     */
+    public function authorize()
```
same

LucasLeandro1204 commented on app/Http/Requests/StoreTask.php:
```diff
@@ -0,0 +1,31 @@
+     *
+     * @return array
+     */
+    public function rules()
```
same

LucasLeandro1204 commented on app/Http/Requests/StoreTask.php:
```diff
@@ -0,0 +1,31 @@
+     *
+     * @return bool
+     */
+    public function authorize()
```
same

LucasLeandro1204 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -30,35 +37,33 @@ public function show(Task $task): Task
-     * Delete a task
+     * Delete an existing task.
+     *
+     * @param Task $task
```
you can drop these annotations too

LucasLeandro1204 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -2,8 +2,9 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\StoreTask;
```
Please, order imports by length > alphabetically

LucasLeandro1204 reviewed (changes requested):
Hey, thank you for your contribution =)

Requested some changes, mostly (if not they all) are related to annotations vs typehint

nickdavies791 committed:
Use type hints instead of annotation

Replaces all annotations in docblocks with the type hint. Restores
:memory: as phpunit database.

nickdavies791 commented:
Thanks. Saw the project and thought I'd try forking and updating for the first time!

I've made the requested changes and pushed in another commit.

LucasLeandro1204 commented:
Haha thank you bro, great contribution =)
